docs/lang/articles/c-api/taichi_core.md

```diff
@@ -0,0 +1,1246 @@
+
+```cpp
+TiRuntime runtime = ti_create_runtime(TI_ARCH_VULKAN);
+```
+
+When your program runs to the end, ensure that:
+- You destroy the runtime instance,
+- All related resources are destroyed before the [`TiRuntime`](#handle-tiruntime) itself.
+
+```cpp
+ti_destroy_runtime(runtime);
+```
+
+### Allocate and free memory
+
+Allocate a piece of memory that is visible only to the device. On the GPU backends, it usually means that the memory is located in the graphics memory (GRAM).
+
+```cpp
+TiMemoryAllocateInfo mai {};
+mai.size = 1024; // Size in bytes.
+mai.usage = TI_MEMORY_USAGE_STORAGE_BIT;
+TiMemory memory = ti_allocate_memory(runtime, &mai);
+```
+
+Allocated memory is automatically freed when the related [`TiRuntime`](#handle-tiruntime) is destroyed. You can also manually free the allocated memory.
+
+```cpp
+ti_free_memory(runtime, memory);
+```
+
+### Allocate host-accessible memory
+
+By default, memory allocations are physically or conceptually local to the offload target for performance reasons. You can configure the [`TiMemoryAllocateInfo`](#structure-timemoryallocateinfo) to enable host access to memory allocations. But please note that host-accessible allocations *may* slow down computation on GPU because of the limited bus bandwidth between the host memory and the device.
+
+You *must* set `host_write` to [`TI_TRUE`](#definition-ti_true) to allow zero-copy data streaming to the memory.
+
+```cpp
+TiMemoryAllocateInfo mai {};
+mai.size = 1024; // Size in bytes.
+mai.host_write = TI_TRUE;
+mai.usage = TI_MEMORY_USAGE_STORAGE_BIT;
+TiMemory steaming_memory = ti_allocate_memory(runtime, &mai);
+
+// ...
+
+std::vector<uint8_t> src = some_random_data_source();
+
+void* dst = ti_map_memory(runtime, steaming_memory);
+std::memcpy(dst, src.data(), src.size());
+ti_unmap_memory(runtime, streaming_memory);
+```
+
+To read data back to the host, `host_read` *must* be set to [`TI_TRUE`](#definition-ti_true).
+
+```cpp
+TiMemoryAllocateInfo mai {};
+mai.size = 1024; // Size in bytes.
+mai.host_read = TI_TRUE;
+mai.usage = TI_MEMORY_USAGE_STORAGE_BIT;
+TiMemory read_back_memory = ti_allocate_memory(runtime, &mai);
+
+// ...
+
+std::vector<uint8_t> dst(1024);
+void* src = ti_map_memory(runtime, read_back_memory);
+std::memcpy(dst.data(), src, dst.size());
+ti_unmap_memory(runtime, read_back_memory);
+
+ti_free_memory(runtime, read_back_memory);
+```
+
+> You can set `host_read` and `host_write` at the same time.
+
+### Load and destroy a Taichi AOT module
+
+You can load a Taichi AOT module from the filesystem.
+
+```cpp
+TiAotModule aot_module = ti_load_aot_module(runtime, "/path/to/aot/module");
+```
+
+`/path/to/aot/module` should point to the directory that contains a `metadata.tcb`.
+
+You can destroy an unused AOT module if you have done with it; but please ensure there is no kernel or compute graph related to it pending to [`ti_submit`](#function-ti_submit).
```

Comment:
```suggestion
You can destroy an unused AOT module, but please ensure that there is no kernel or compute graph related to it pending to [`ti_submit`](#function-ti_submit).
```
- Am not quite certain about "If you have done with it". Grammatically it ought to be "if you are done with it". I believe it is a given, so I omitted it. 
- Do not use conj. together with semicolons.